<launch>
    <arg name="debug" default="false"/>

    <param name="robot_description" command="$(find xacro)/xacro.py '$(find pr2_description)/robots/pr2.urdf.xacro'"/>

    <env name="CPUPROFILE" value="prof.out"/>

    <arg unless="$(arg debug)" name="launch-prefix" value=""/>
    <arg     if="$(arg debug)" name="launch-prefix" value="gdb ex --args"/>

    <env name="SMPL_VISUALIZE_CONFIG_FILE" value="$(find smpl_test)/config/smpl_visual.ini"/>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find smpl_test)/config/rosconsole.properties"/>

    <node pkg="smpl_test" type="test_ara_bfs_jsl" name="test_ara_bfs_jsl" launch-prefix="$(arg launch-prefix)">
        <param name="planning_frame" value="odom_combined"/>

        <!-- 1. robot model (for planning) -->
        <rosparam param="robot_model">
            group_name: right_arm
            kinematics_frame:
                torso_lift_link
            chain_tip_link:
                r_gripper_palm_link
        </rosparam>

        <!-- collision checking -->
        <rosparam command="load" file="$(find sbpl_collision_checking_test)/config/collision_model_pr2.yaml"/>

        <!-- planner params -->
        <rosparam command="load" file="$(find smpl_test)/config/pr2_right_arm.yaml"/>
        <param name="planning/mprim_filename" value="$(find smpl_test)/config/pr2.mprim"/>
        <param name="allowed_planning_time" value="15.0"/>

        <!-- 2. scene -->
        <!-- 2.1 robot state -->
        <rosparam command="load" file="$(find smpl_test)/experiments/pr2_goal.yaml"/>

        <!-- 2.2 world state -->
        <param name="object_filename" value="$(find smpl_test)/env/tabletop.env"/>

        <!-- 3. scenario -->
    </node>
</launch>

